### ISSUE TITLE
Index not being used for some Match queries.

### ISSUE THREAD

**Issue by schwab** [FalkorDB/FalkorDB#796]
## TLDR:
Vastly different query plans for :
```

graph.profile ... CYPHER MATCH (x:typeA {hash_value:"999"}), (y:typeB hash_value:"222"}) RETURN id(x), id(y)'

VS

graph.profile ... 'CYPHER MATCH (x:typeA {hash_value:"999"}) with x, MATCH (y:typeB {hash_value:"222"}) RETURN id(f), id(d)'
```

We have a graph with indexes set on the hash_value field for 2 node types like this:
```
...
10) 1) "dim_demographics"
      2) "[hash_value]"
      3) "{hash_value: [RANGE]}"
      4) "english"
      5) "[]"
      6) "NODE"
      7) "OPERATIONAL"
      8) "{gcPolicy: 0, score: 1.000000, lang: english, fields: [{path: range:hash_value, name: range:hash_value, options: 0, textWeight: 0.000000, tagCaseSensitive: true}, {path: NONE_INDEXABLE_FIELDS, name: NONE_INDEXABLE_FIELDS, options: 0, textWeight: 0.000000, tagCaseSensitive: true}], numDocuments: 819456, maxDocId: 819456, docTableSize: 60639744, sortablesSize: 0, docTrieSize: 25672192, numTerms: 0, numRecords: 819456, invertedSize: 819456, invertedCap: 0, skipIndexesSize: 0, scoreIndexesSize: 0, offsetVecsSize: 0, offsetVecRecords: 0, termsSize: 0, indexingFailures: 0, totalCollected: 0, numCycles: 0, totalMSRun: 0, lastRunTimeMs: 0}"
      ...
16) 1) "fact_clinics_advisor_sch"
      2) "[hash_value]"
      3) "{hash_value: [RANGE]}"
      4) "english"
      5) "[]"
      6) "NODE"
      7) "OPERATIONAL"
      8) "{gcPolicy: 0, score: 1.000000, lang: english, fields: [{path: range:hash_value, name: range:hash_value, options: 0, textWeight: 0.000000, tagCaseSensitive: true}, {path: NONE_INDEXABLE_FIELDS, name: NONE_INDEXABLE_FIELDS, options: 0, textWeight: 0.000000, tagCaseSensitive: true}], numDocuments: 3117546, maxDocId: 3117546, do
...
```
However the indexes are not being used for this query:
```
graph.profile graph:clinics_advisor_sch 'CYPHER MATCH (f:fact_clinics_advisor_sch {hash_value:"a95c68ce767f3fdcb637623919dc2841"}),(d:dim_demographics {hash_value:"a226cb12fc2286fcd68db8307f1c5424"}) RETURN id(f), id(d)'

1) "Results | Records produced: 1, Execution time: 0.000981 ms"
2) "    Project | Records produced: 1, Execution time: 0.003376 ms"
3) "        Value Hash Join | fact_hash = property(f,hash_value,3) | Records produced: 1, Execution time: 392.636555 ms"
4) "            Value Hash Join | property(d,hash_value,3) = dim_hash | Records produced: 1, Execution time: 1273.042005 ms"
5) "                Node By Label Scan | (d:dim_demographics) | Records produced: 819456, Execution time: 198.311692 ms"
6) "                Project | Records produced: 1, Execution time: 0.003447 ms"
7) "                    Unwind | Records produced: 1, Execution time: 0.017884 ms"
8) "            Node By Label Scan | (f:fact_clinics_advisor_sch) | Records produced: 3117546, Execution time: 192.369517 ms"
```
This leads to very long runtime for the query.

However we found that if you change the query to this it uses the index and the queries are very fast.  In this case w went from almost 2 seconds to < 0.01ms! 
:
```
graph.profile graph:clinics_advisor_sch 'CYPHER MATCH (f:fact_clinics_advisor_sch) where f.hash_value = "a95c68ce767f3fdcb637623919dc2841" with f,  MATCH (d:dim_demographics) where d.hash_value = "a226cb12fc2286fcd68db8307f1c5424" RETURN id(f), id(d)'

1) "Results | Records produced: 0, Execution time: 0.000261 ms"
2) "    Project | Records produced: 0, Execution time: 0.000140 ms"
3) "        Filter | Records produced: 0, Execution time: 0.000150 ms"
4) "            Node By Label Scan | (d:dim_demographics) | Records produced: 0, Execution time: 0.000471 ms"
5) "                Project | Records produced: 0, Execution time: 0.000120 ms"
6) "                    Filter | Records produced: 0, Execution time: 0.000391 ms"
7) "                        Node By Label Scan | (f:fact_clinics_advisor_sch) | Records produced: 0, Execution time: 0.000070 ms"
```

The only real change is using 2 MATCH statements instead of connecting the node statements with a comma. The results are identical, but the query plan is very different. Why are these not both performing the same query plan?

**Comment by AviAvni** [comment]
what version are you using?
here is my try with latest version
```
127.0.0.1:6379> graph.profile x 'CYPHER MATCH (f:fact_clinics_advisor_sch {hash_value:"a95c68ce767f3fdcb637623919dc2841"}),(d:dim_demographics {hash_value:"a226cb12fc2286fcd68db8307f1c5424"}) RETURN id(f), id(d)'
1) "Results | Records produced: 0, Execution time: 0.010166 ms"
2) "    Project | Records produced: 0, Execution time: 0.008917 ms"
3) "        Cartesian Product | Records produced: 0, Execution time: 0.009709 ms"
4) "            Node By Index Scan | (f:fact_clinics_advisor_sch) | Records produced: 0, Execution time: 0.228791 ms"
5) "            Node By Index Scan | (d:dim_demographics) | Records produced: 0, Execution time: 0.000000 ms"
```

**Comment by schwab** [comment]
Thank you for pointing out these performance improvements.  We are definitely behind on our releases so we'll get that updated!
